lib/effects/sequence_effect.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:meta/meta.dart';
+
+import './effects.dart';
+import '../position.dart';
+
+class SequenceEffect extends PositionComponentEffect {
+  final List<PositionComponentEffect> effects;
+  int _currentIndex = 0;
+  PositionComponentEffect currentEffect;
+  bool _currentWasAlternating;
+  double _driftModifier = 0.0;
+
+  SequenceEffect({
+    @required this.effects,
+    isInfinite = false,
+    isAlternating = false,
+  }) : super(isInfinite, isAlternating);
+
+  @override
+  set component(_comp) {
+    super.component = _comp;
+    final originalSize = _comp.toSize();
+    final originalPosition = _comp.toPosition();
+    Position currentSize = _comp.toSize();
+    Position currentPosition = _comp.toPosition();
+    effects.forEach((effect) {
+      effect.reset();
+      _comp.setBySize(currentSize);
+      _comp.setByPosition(currentPosition);
+      effect.component = _comp;
+      if (effect is MoveEffect) {
+        currentPosition = effect.destination;
+      } else if (effect is ScaleEffect) {
+        currentSize = Position.fromSize(effect.size);
+      }
```

Comment:
what if it's something else? should we throw an error in case we use an unsupported effect? I am also wondering if there is a way to make SequenceEffect generic so it supports any sort of PositionComponentEffect, even custom ones. eg can't we call `component` on the `currentEffect` instead of if/else casing all the possible instances?

Reply:
It can support the rotate effect too which is not covered by that clause. The problem is that scale and move is absolute meanwhile the rotate is relative. That is probably a coming addition too, that all can be both relative and absolute. Not sure how to solve the casing though, open to suggestions.

Reply:
Fixed this by moving the end properties to the abstract class so no more type specific initialization. Let me know what you think.